cohttp-async/src/cohttp_async.ml

```diff
@@ -1,11 +1,8 @@
 module Body = Body
 module Body_raw = Body [@@deprecated "Use Body"]
 module Client = Client
-module Io = Io [@@deprecated "This module is not for public consumption"]
+module Io = Io.IO
+module Input_channel = Input_channel
 module Request = Cohttp.Request [@@deprecated "Use Cohttp.Request directly"]
 module Response = Cohttp.Response [@@deprecated "Use Cohttp.Response directly"]
 module Server = Server
-
-module Private = struct
```

Comment:
Why you needed to make this exposed again? It does not seem to make a difference for the implementation and can be still referred to as a private module

Reply:
Similar to the IO module. This gets passed to the expert response callback so it probably is better to keep it part of the public api?

Reply:
I ended up taking a different approach. The only place we need this is for the expert response within the servers, so I've moved a restricted view of Input_channel within the server module.